### PR TITLE
Set up Sonar integration

### PR DESCRIPTION
**Pull request by kruai**
## Overview

This PR is being created to address [this Jira](https://issues.redhat.com/browse/RHCLOUD-10357).
It adds Sonar integration functionality, and adds setup instructions to the README.

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [ ] ~~Input Validation~~
- [ ] ~~Output Encoding~~
- [ ] ~~Authentication and Password Management~~
- [ ] ~~Session Management~~
- [ ] ~~Access Control~~
- [ ] ~~Cryptographic Practices~~
- [ ] ~~Error Handling and Logging~~
- [ ] ~~Data Protection~~
- [ ] ~~Communication Security~~
- [ ] ~~System Configuration~~
- [ ] ~~Database Security~~
- [ ] ~~File Management~~
- [ ] ~~Memory Management~~
- [ ] ~~General Coding Practices~~

### CODE REVIEW

**Comment by kruai** [comment]
I was going to mark this as WIP until I got some sort of git hook set up, but SonarScanner takes too long to be a hook. No project has yet been able to set up Sonar as a PR check, so I've created this Jira to track that work: https://issues.redhat.com/browse/RHCLOUD-11091

File: README.md [hunk 1]
```diff
@@ -67,6 +67,18 @@ pytest test_json_validators.py
 Depending on the environment, it might be necessary to set the DB related environment
 variables (INVENTORY_DB_NAME, INVENTORY_DB_HOST, etc).
 
+## Sonar Integration
+
+This project uses Sonar for static code analysis. To set up sonar-scanner on your machine, [download the CLI](https://sonarqube.corp.redhat.com/documentation/analysis/scan/sonarscanner/) and set the `SONAR_PATH` env var like this:
+
+``` sh
+SONAR_PATH=/path/to/sonar-scanner/lib/sonar-scanner-cli-${SONAR_CLI_VERSION_NUMBER}.jar
```
Comment (ryandillinfelton) [inline README.md:75]: I had to use the `export` command here to get the env var set

Comment (kruai) [reply README.md]: Got it! Pushed a fix.

File: sonar_scan.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Error out if sonar scanner isn't installed and configured properly
+if [[ -z "${SONAR_PATH}" ]]
+then
+    echo "Scanner jar required! Download it from here https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-4.5.0.2216.zip, and then set the env var with SONAR_PATH=/path/to/sonar-scanner/lib/sonar-scanner-cli-4.5.0.2216.jar"
```
Comment (ryandillinfelton) [inline sonar_scan.sh:4]: if you updated the ENV var command in the other spot I think it would be good to update it here as well for consistency

**Review by ryandillinfelton — DISMISSED** [review]
I recommended a small wording change. LGTM otherwise :+1:

File: sonar_scan.sh (hunk 2 above)
Comment (kruai) [reply sonar_scan.sh]: Done!

**Review by ryandillinfelton — APPROVED** [review]

**Pull request merged**